src/config.c

```diff
@@ -2056,92 +2080,73 @@ static int isValidProcTitleTemplate(char *val, const char **err) {
     return 1;
 }
 
-static int updateProcTitleTemplate(char *val, char *prev, const char **err) {
-    UNUSED(val);
-    UNUSED(prev);
+static int applyProcTitleTemplate(const char **err) {
     if (redisSetProcTitle(NULL) == C_ERR) {
         *err = "failed to set process title";
         return 0;
     }
     return 1;
 }
 
-static int updateHZ(long long val, long long prev, const char **err) {
-    UNUSED(prev);
+static int applyHZ(const char **err) {
     UNUSED(err);
     /* Hz is more a hint from the user, so we accept values out of range
      * but cap them to reasonable values. */
-    server.config_hz = val;
     if (server.config_hz < CONFIG_MIN_HZ) server.config_hz = CONFIG_MIN_HZ;
     if (server.config_hz > CONFIG_MAX_HZ) server.config_hz = CONFIG_MAX_HZ;
```

Comment:
Good question but as far as I can think of it's a valid approach. This is the only way to cap the value implicitly and not return an out of range error.

Reply:
for the record, i do have a plan to clamp many configs to more appropriate ranges, but i don't wanna introduce a breaking change (don't wanna error in case someone currently passes an insane value).
so some day soon, i'd like to introduce some automatic clamping flag to config.c.

till then, i just wanted to be sure these two lines don't violate anything of the new approach.

Reply:
Even if we add some auto clamping mechanism it'll just make this redundant, no need to worry about it now.